utils/splunk.py

```diff
@@ -36,8 +49,48 @@ def get_auth_tuple(self):
         return self.username, self.password
 
 
+class SavedSearches(object):
+
+    def __init__(self, saved_searches):
+        self.searches = filter(lambda ss: ss.name is not None, saved_searches)
+        self.matches = filter(lambda ss: ss.match is not None, saved_searches)
+        self.log = logging.getLogger('%s.%s' % (__name__, "SavedSearches"))
+
+    def update_searches(self, saved_searches):
+        """
+        :param saved_searches: List of strings with names of observed saved searches
+        """
+        # Drop missing matches
+        self.searches = filter(lambda s: s.match is None or s.name in saved_searches, self.searches)
+
+        # Filter already instantiated searches
+        new_searches = set(saved_searches).difference([s.name for s in self.searches])
+
+        # Match new searches
+        for new_search in new_searches:
+            for match in self.matches:
+                if re.match(match.match, new_search) is not None:
+                    search = copy.deepcopy(match)
+                    search.name = new_search
+                    self.log.info("Created new saved search on saved search '%s'" % new_search)
```

Comment:
I'd prefer a text change here. Something along the lines of "Adding saved search '%s'"?